docs/wip/sonarqube.md

```diff
@@ -1,5 +1,42 @@
-## Annotations and when should the checks fail
+## Sonarqube
 
-## How to setup sonarqube on CI
+All repositories using VTEX Store Framework will have **SonarQube analysis enabled by default**. This means that every opened Pull Request's code will be **analysed using quality and security standards,** preventing bugs and bad practices going into production. 
 
+The SonarQube app will annotate lines of code on the PR using the analysis's response. There are two types of annotations:
+- **Warning:** indicates an improvement point.
+- **Failure:** indicates a potential compile error or security vulnerability.
 
+### Examples:
+<p align="center">
+  <img alt="SonarQube warning" src="https://user-images.githubusercontent.com/18706156/103794175-2c2a9e00-5023-11eb-98bc-cc4b9a3c7b67.png" width="500"/>
+</p>
+<p align="center">
+  <i>Warning due a Typescript non-null assertion.</i>
+</p>
+
+<p align="center">
+  <img alt="SonarQube failure" src="https://user-images.githubusercontent.com/18706156/103794365-65630e00-5023-11eb-87f1-1ebf6c7e2eea.png" width="500"/>
+</p>
+<p align="center">
+  <i>Failure due an import from an unidentified module.</i>
+</p>
+
+## Pull Request checks
+
+Besides notifying the PR's author with the annotations, the SonarQube integration also **adds a check on the Pull Request** with either `success` or `failure`:
+
+<p align="center">
+  <img alt="PR checks" src="https://user-images.githubusercontent.com/18706156/103794852-ffc35180-5023-11eb-8fd7-f7a998de5306.png" width="500"/>
+</p>
+
+The check will fail if there is **at least one failure annotation**, blocking the PR from merging into the `main` branch.
+
+## Antecipating the rule checkings
+
+SonarQube ueses [ESLint](https://eslint.org/) to analyse the source code and come up with the annotations. Since our repositories are configured to use ESLint as a linter, most issues should be easy to spot on the code editor. Make sure to run `yarn` to install the dependencies.
```

Comment:
```suggestion
SonarQube ueses [ESLint](https://eslint.org/) to analyze the source code and come up with the annotations. 

This way, you can anticipate possible SonarQube issues by installing the ESLint. For that, run the `yarn` command on your project. This will ensure you install all SFJ dependencies, including the ESLint.

In the following, since SFJ repositories use ESLint as a linter, most issues will be apparent in the code editor.  
```